src/solver/libModelObject/include/antares/solver/libObjectModel/model.h

```diff
@@ -39,22 +39,61 @@ namespace Antares::Solver::ObjectModel
 class Model
 {
 public:
-    Model();
-    ~Model() = default;
+    const std::string& Id() const
+    {
+        return id_;
+    }
 
-    std::vector<Constraint*> getConstraints();
+    Expression Objective() const
+    {
+        return objective_;
+    }
+
+    const std::map<std::string, Constraint>& getConstraints() const
+    {
+        return constraints_;
+    }
+
+    const std::map<std::string, Parameter>& Parameters() const
+    {
+        return parameters_;
+    }
+
+    const std::map<std::string, Variable>& Variables() const
+    {
+        return variables_;
+    }
+
+    const std::map<std::string, Port>& Ports() const
+    {
+        return ports_;
+    }
 
 private:
+    friend class ModelBuilder;
     std::string id_;
     Expression objective_;
 
     std::map<std::string, Parameter> parameters_;
     std::map<std::string, Variable> variables_;
-
     std::map<std::string, Constraint> constraints_;
-    std::map<std::string, Constraint> bindingConstraints_;
-
     std::map<std::string, Port> ports_;
 };
 
+class ModelBuilder
+{
+public:
+    ModelBuilder& withId(std::string_view id);
+    ModelBuilder& withObjective(Expression objective);
+    ModelBuilder& withParameters(std::vector<Parameter>& parameters);
+    ModelBuilder& withVariables(std::vector<Variable>& variables);
+    ModelBuilder& withPorts(std::vector<Port>& ports);
```

Comment:
you may make it even easier to use, if you implement withParameter(Parameter& parameter), withVariable, and withPort, so the user can add objects iteratively.
Better yet, you can create a ParameterBuilder, a VariableBuilder, and a PortBuilder (see my previous comment for an example)

Reply:
I don't think there is value to this at the moment.
The current interface stem from the usage where models are created by parsing yaml, not built be hand.
At least I don't think a value by value design would plug in nicely. If you have a example I'm open to it